lib/ae_mdw_web/controllers/channel_controller.ex

```diff
@@ -22,10 +22,22 @@ defmodule AeMdwWeb.ChannelController do
   end
 
   @spec channel(Conn.t(), map()) :: Conn.t()
-  def channel(%Conn{assigns: %{state: state}} = conn, %{"id" => id}) do
+  def channel(%Conn{assigns: %{state: state}} = conn, %{"id" => id} = params) do
+    block_hash = params["block_hash"]
+
     with {:ok, channel_pk} <- Validate.id(id, [:channel]),
-         {:ok, channel} <- Channels.fetch_channel(state, channel_pk) do
+         true <- valid_optional_block_hash?(block_hash),
```

Comment:
This is the first decoding that we do, we can use its value by returning `{:ok, block_hash}`. In addition, if the parameter is not a valid `block_hash` the whole function returns `false` which is not handled by the FallbackController and it returns a 500 error

Reply:
done